base/strings/util.jl

```diff
@@ -77,18 +77,23 @@ chop(s::AbstractString) = SubString(s, 1, endof(s)-1)
     chomp(s::AbstractString)
 
 Remove a single trailing newline from a string.
+Lines in the input can end in `'\\n'`, `'\\r'`, or `'\\r\\n'`.
 """
 function chomp(s::AbstractString)
     i = endof(s)
-    if (i < 1 || s[i] != '\n') return SubString(s, 1, i) end
+    if (i < 1 || (s[i] != '\n' &&  s[i] != '\r')) return SubString(s, 1, i) end
+    if (s[i] == '\r') return SubString(s, 1, i-1) end
     j = prevind(s,i)
-    if (j < 1 || s[j] != '\r') return SubString(s, 1, i-1) end
+    if (j < 1 || (s[j] != '\r' && s[i] == '\n')) return SubString(s, 1, i-1) end
     return SubString(s, 1, j-1)
 end
+
 function chomp(s::String)
     i = endof(s)
-    if i < 1 || s.data[i] != 0x0a
+    if i < 1 || (s.data[i] != 0x0a && s.data[i] != 0x0d)
```

Comment:
Since you're making the function more complex, I would find it clearer to replace the integer codes with `UInt8('\n')` and  `UInt8('\r')`. Same in `readline`.